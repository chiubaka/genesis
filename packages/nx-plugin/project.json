{
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "packages/nx-plugin/src",
  "projectType": "library",
  "targets": {
    "build": {
      "executor": "@nrwl/js:tsc",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/packages/nx-plugin",
        "main": "packages/nx-plugin/src/index.ts",
        "tsConfig": "packages/nx-plugin/tsconfig.lib.json",
        "updateBuildableProjectDepsInPackageJson": false,
        "assets": [
          "packages/nx-plugin/*.md",
          {
            "input": "./packages/nx-plugin/src",
            "glob": "**/!(*.ts)",
            "output": "./src"
          },
          {
            "input": "./packages/nx-plugin/src",
            "glob": "**/*.d.ts",
            "output": "./src"
          },
          {
            "input": "./packages/nx-plugin",
            "glob": "generators.json",
            "output": "."
          },
          {
            "input": "./packages/nx-plugin",
            "glob": "executors.json",
            "output": "."
          }
        ]
      }
    },
    "publish:local": {
      "executor": "@chiubaka/nx-plugin:local-publish",
      "dependsOn": [{ "target": "build", "projects": "self" }],
      "options": {
        "registryUrl": "http://localhost:4873",
        "packagePath": "dist/packages/nx-plugin"
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["packages/nx-plugin"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": [
        "reports/coverage/packages/nx-plugin",
        "reports/junit/nx-plugin.xml"
      ],
      "options": {
        "jestConfig": "packages/nx-plugin/jest.config.ts",
        "passWithNoTests": true
      }
    }
  },
  "tags": []
}
