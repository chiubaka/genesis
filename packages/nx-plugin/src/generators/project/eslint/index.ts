export * from "./eslintProjectGenerator";
export { eslintProjectGenerator as default } from "./eslintProjectGenerator";
