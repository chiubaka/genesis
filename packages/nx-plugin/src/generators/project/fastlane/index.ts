export * from "./fastlaneProjectGenerator";
export * from "./fastlaneProjectGenerator.schema";
