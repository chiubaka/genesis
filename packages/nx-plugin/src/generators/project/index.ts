export * from "./eslint";
export * from "./jest";
export * from "./readme";
