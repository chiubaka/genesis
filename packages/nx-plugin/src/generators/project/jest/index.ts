export * from "./jestProjectGenerator";
export { jestProjectGenerator as default } from "./jestProjectGenerator";
