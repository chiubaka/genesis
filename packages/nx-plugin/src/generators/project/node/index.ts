export * from "./nodeProjectGenerator";
export { nodeProjectGenerator as default } from "./nodeProjectGenerator";
