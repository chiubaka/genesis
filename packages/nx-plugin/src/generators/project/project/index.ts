export * from "./projectGenerator";
export * from "./projectGenerator.schema";
