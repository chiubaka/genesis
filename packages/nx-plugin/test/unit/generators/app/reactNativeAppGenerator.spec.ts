import { createTreeWithEmptyWorkspace } from "@chiubaka/nx-plugin-testing";
import {
  ProjectConfiguration,
  readJson,
  TargetConfiguration,
  Tree,
} from "@nx/devkit";

import {
  EsLintExecutorOptions,
  PackageJson,
  Project,
  reactNativeAppGenerator,
} from "../../../../src";
import { fileMatchesSnapshot } from "../../../cases";
import { projectTestCases } from "../../../cases/project/projectTestCases";

describe("reactNativeAppGenerator", () => {
  let tree: Tree;
  let project: Project;
  let e2eProject: Project;

  const getProject = () => {
    return project;
  };

  const getE2eProject = () => {
    return e2eProject;
  };

  beforeAll(() => {
    tree = createTreeWithEmptyWorkspace();
    project = new Project(tree, "react-native-app", "application");
    e2eProject = new Project(tree, "react-native-app-e2e", "e2e");
  });

  projectTestCases(getProject);

  it("removes the React Native app from yarn workspaces", () => {
    const packageJson = readJson<PackageJson>(tree, "package.json");

    const workspaces = (packageJson.workspaces ?? { packages: [] }) as {
      packages: string[];
    };

    const packages = workspaces.packages;

    expect(packages).toContain(`!${project.path()}`);
  });

  it("moves the test-setup.ts file to test/setup/setup.ts", () => {
    // eslint-disable-next-line security/detect-non-literal-fs-filename
    expect(tree.exists(project.path("test-setup.ts"))).toBe(false);
    // eslint-disable-next-line security/detect-non-literal-fs-filename
    expect(tree.exists(project.testPath("setup/setup.ts"))).toBe(true);
  });

  beforeAll(async () => {
    await reactNativeAppGenerator(tree, {
      name: "react-native-app",
      displayName: "React Native App",
      appId: "com.chiubaka.ReactNativeApp",
    });
  });

  describe("README", () => {
    it("adds first-time setup instructions for Android SDK", () => {
      expect(tree).toHaveFileWithContent(
        project.path("README.md"),
        "Android SDK",
      );
    });

    it("adds first-time setup instructions for iOS and XCode SDK", () => {
      expect(tree).toHaveFileWithContent(project.path("README.md"), "iOS SDK");
    });

    it("adds first-time setup instructions for emulators and Detox configuration", () => {
      expect(tree).toHaveFileWithContent(
        project.path("README.md"),
        ".detoxrc.json",
      );
    });

    it("adds first-time setup instructions for ruby", () => {
      expect(tree).toHaveFileWithContent(project.path("README.md"), "ruby");
    });

    // Unless I can accomplish this with the Podfile automatically
    // Should include info about Fastlane and getting Fastlane to generate certificates and profiles for you
    it.todo("adds first-time setup instructions for iOS code signing");

    // Google Play first-time setup: https://docs.fastlane.tools/actions/supply/
    it.todo("adds first-time setup instructions for Fastlane");
  });

  describe("Gemfile", () => {
    fileMatchesSnapshot("Gemfile", getProject, (project: Project) => {
      return project.path("Gemfile");
    });

    it("updates the required ruby version", () => {
      expect(tree).toHaveFileWithContent(
        project.path("Gemfile"),
        'ruby ">= 3.2.2"',
      );
    });

    it("replaces single quotes with double quotes", () => {
      expect(tree).not.toHaveFileWithContent(project.path("Gemfile"), "'");
    });
  });

  describe("native projects", () => {
    describe.skip("fastlane", () => {
      it("adds fastlane to the Gemfile", () => {
        expect(tree).toHaveFileWithContent(
          project.path("Gemfile"),
          'gem "fastlane"',
        );
      });

      fileMatchesSnapshot("Fastfile", getProject, (project: Project) => {
        return project.path("fastlane/Fastfile");
      });

      describe.skip("Appfile", () => {
        fileMatchesSnapshot("Appfile", getProject, (project: Project) => {
          return project.path("fastlane/Appfile");
        });

        it.todo("fills in a value for app_identifier");

        it.todo("fills in a value for apple_id");

        it.todo("fills in a value for itc_team_id");

        it.todo("fills in a value for team_id");

        it.todo("fills in a value for json_key_file");

        it.todo("fills in a value for package_name");
      });
    });

    describe("ios", () => {
      let iosXcodeProjectPath: string;

      beforeAll(() => {
        iosXcodeProjectPath = project.path(
          "ios/ReactNativeApp.xcodeproj/project.pbxproj",
        );
      });

      it("updates the bundle identifier", () => {
        expect(tree).not.toHaveFileWithContent(
          iosXcodeProjectPath,
          "org.reactjs.native.example.$(PRODUCT_NAME:rfc1034identifier)",
        );

        expect(tree).toHaveFileWithContent(
          iosXcodeProjectPath,
          "com.chiubaka.ReactNativeApp",
        );
      });

      // Aspirational! But not possible to actually accomplish unless I add stronger
      // capabilities to understand and modify the generated .pbxproj file
      // https://linear.app/chiubaka/issue/CT-723/update-the-development-team-in-the-generated-ios-pbxproj-file
      it.skip("updates the development team ID", () => {
        expect(tree).toHaveFileWithContent(
          iosXcodeProjectPath,
          'DEVELOPMENT_TEAM = "ABC1234567"',
        );
      });

      it("patches the Start Packager build phase to provide the correct PROJECT_ROOT", () => {
        expect(tree).toHaveFileWithContent(
          iosXcodeProjectPath,
          "export PROJECT_ROOT=${SRCROOT}",
        );
      });

      describe("Podfile", () => {
        fileMatchesSnapshot("Podfile", getProject, (project: Project) => {
          return project.path("ios/Podfile");
        });

        it("replaces single quotes with double quotes", () => {
          expect(tree).not.toHaveFileWithContent(
            project.path("ios/Podfile"),
            "target 'ReactNativeApp' do",
          );
        });
      });
    });

    describe("android", () => {
      describe("build.gradle", () => {
        it("updates the namespace", () => {
          expect(tree).not.toHaveFileWithContent(
            project.path("android/app/build.gradle"),
            'namespace "com.reactnativeapp"',
          );

          expect(tree).toHaveFileWithContent(
            project.path("android/app/build.gradle"),
            'namespace "com.chiubaka.reactnativeapp"',
          );
        });

        it("updates the applicationId", () => {
          expect(tree).not.toHaveFileWithContent(
            project.path("android/app/build.gradle"),
            'applicationId "com.reactnativeapp"',
          );

          expect(tree).toHaveFileWithContent(
            project.path("android/app/build.gradle"),
            'applicationId "com.chiubaka.ReactNativeApp"',
          );
        });
      });

      describe("java files", () => {
        describe("moves java source files to appropriate new package directories", () => {
          it("androidTest", () => {
            expect(tree).toHaveRenamedFile(
              project.path(
                "android/app/src/androidTest/java/com/reactnativeapp",
              ),
              project.path(
                "android/app/src/androidTest/java/com/chiubaka/reactnativeapp",
              ),
            );
          });

          it("debug", () => {
            expect(tree).toHaveRenamedFile(
              project.path("android/app/src/debug/java/com/reactnativeapp"),
              project.path(
                "android/app/src/debug/java/com/chiubaka/reactnativeapp",
              ),
            );
          });

          it("main", () => {
            expect(tree).toHaveRenamedFile(
              project.path("android/app/src/main/java/com/reactnativeapp"),
              project.path(
                "android/app/src/main/java/com/chiubaka/reactnativeapp",
              ),
            );
          });

          it("release", () => {
            expect(tree).toHaveRenamedFile(
              project.path("android/app/src/release/java/com/reactnativeapp"),
              project.path(
                "android/app/src/release/java/com/chiubaka/reactnativeapp",
              ),
            );
          });
        });

        describe("updates the package for moved java source files", () => {
          it("DetoxTest.java", () => {
            const detoxTestPath = project.path(
              "android/app/src/androidTest/java/com/chiubaka/reactnativeapp/DetoxTest.java",
            );

            expect(tree).not.toHaveFileWithContent(
              detoxTestPath,
              "package com.reactnativeapp",
            );

            expect(tree).toHaveFileWithContent(
              detoxTestPath,
              "package com.chiubaka.reactnativeapp",
            );
          });

          it("debug ReactNativeFlipper.java", () => {
            const reactNativeFlipperPath = project.path(
              "android/app/src/debug/java/com/chiubaka/reactnativeapp/ReactNativeFlipper.java",
            );

            expect(tree).not.toHaveFileWithContent(
              reactNativeFlipperPath,
              "package com.reactnativeapp",
            );

            expect(tree).toHaveFileWithContent(
              reactNativeFlipperPath,
              "package com.chiubaka.reactnativeapp",
            );
          });

          it("release ReactNativeFlipper.java", () => {
            const reactNativeFlipperPath = project.path(
              "android/app/src/release/java/com/chiubaka/reactnativeapp/ReactNativeFlipper.java",
            );

            expect(tree).not.toHaveFileWithContent(
              reactNativeFlipperPath,
              "package com.reactnativeapp",
            );

            expect(tree).toHaveFileWithContent(
              reactNativeFlipperPath,
              "package com.chiubaka.reactnativeapp",
            );
          });

          it("MainActivity.java", () => {
            const mainActivityPath = project.path(
              "android/app/src/main/java/com/chiubaka/reactnativeapp/MainActivity.java",
            );

            expect(tree).not.toHaveFileWithContent(
              mainActivityPath,
              "package com.reactnativeapp",
            );

            expect(tree).toHaveFileWithContent(
              mainActivityPath,
              "package com.chiubaka.reactnativeapp",
            );
          });

          it("MainApplication.java", () => {
            const mainApplicationPath = project.path(
              "android/app/src/main/java/com/chiubaka/reactnativeapp/MainApplication.java",
            );

            expect(tree).not.toHaveFileWithContent(
              mainApplicationPath,
              "package com.reactnativeapp",
            );

            expect(tree).toHaveFileWithContent(
              mainApplicationPath,
              "package com.chiubaka.reactnativeapp",
            );
          });
        });
      });
    });
  });

  describe("e2e", () => {
    it("generates an e2e project in the right place", () => {
      // eslint-disable-next-line security/detect-non-literal-fs-filename
      expect(tree.exists(e2eProject.path())).toBe(true);
    });

    projectTestCases(getE2eProject, {
      skipPackageJson: true,
      skipReadme: true,
      skipTsConfig: true,
      skipJest: true,
    });

    fileMatchesSnapshot(
      ".detoxrc.json",
      getE2eProject,
      (e2eProject: Project) => {
        return e2eProject.path(".detoxrc.json");
      },
    );

    fileMatchesSnapshot(
      "project.json",
      getE2eProject,
      (e2eProject: Project) => {
        return e2eProject.path("project.json");
      },
    );

    describe("typescript", () => {
      describe("tsconfig.json", () => {
        fileMatchesSnapshot(
          "tsconfig.json",
          getE2eProject,
          (e2eProject: Project) => {
            return e2eProject.path("tsconfig.json");
          },
        );

        it("updates references in tsconfig.json to refer to tsconfig.spec.json", () => {
          expect(tree).toHaveFileWithContent(
            e2eProject.path("tsconfig.json"),
            "tsconfig.spec.json",
          );
        });
      });

      fileMatchesSnapshot(
        "tsconfig.spec.json",
        getE2eProject,
        (e2eProject: Project) => {
          return e2eProject.path("tsconfig.spec.json");
        },
      );

      it("does not generate a tsconfig.e2e.json file", () => {
        // eslint-disable-next-line security/detect-non-literal-fs-filename
        expect(tree.exists(e2eProject.path("tsconfig.e2e.json"))).toBe(false);
      });
    });

    it("updates references in the e2e project.json file", () => {
      const projectJson = readJson<ProjectConfiguration>(
        tree,
        e2eProject.path("project.json"),
      );

      const lintTarget = projectJson.targets
        ?.lint as TargetConfiguration<EsLintExecutorOptions>;

      const lintFilePatterns = lintTarget.options?.lintFilePatterns as string[];

      expect(lintFilePatterns).toHaveLength(1);

      expect(lintFilePatterns[0]).toContain(e2eProject.path());
    });

    it("generates a test directory instead of a src directory", () => {
      // eslint-disable-next-line security/detect-non-literal-fs-filename
      expect(tree.exists(e2eProject.srcPath())).toBe(false);
      // eslint-disable-next-line security/detect-non-literal-fs-filename
      expect(tree.exists(e2eProject.testPath())).toBe(true);
    });

    it("still generates a sample e2e test", () => {
      // eslint-disable-next-line security/detect-non-literal-fs-filename
      expect(tree.exists(e2eProject.testPath("e2e/app.spec.ts"))).toBe(true);
    });

    it("moves the test-setup.ts file to test/setup/setup.ts", () => {
      // eslint-disable-next-line security/detect-non-literal-fs-filename
      expect(tree.exists(e2eProject.path("test-setup.ts"))).toBe(false);
      // eslint-disable-next-line security/detect-non-literal-fs-filename
      expect(tree.exists(e2eProject.testPath("setup/setup.ts"))).toBe(true);
    });

    fileMatchesSnapshot(
      "jest.config.json",
      getE2eProject,
      (e2eProject: Project) => {
        return e2eProject.path("jest.config.json");
      },
    );
  });
});
