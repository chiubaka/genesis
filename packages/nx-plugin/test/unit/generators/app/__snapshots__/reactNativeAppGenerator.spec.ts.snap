// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reactNativeAppGenerator .eslintrc.json matches snapshot 1`] = `
{
  "extends": [
    "../../.eslintrc.json",
  ],
  "ignorePatterns": [
    "node_modules",
  ],
  "overrides": [
    {
      "files": [
        "jest.config.ts",
        "test/**/*.ts",
        "*.spec.ts",
        "*.test.ts",
        "test/**/*.tsx",
        "*.spec.tsx",
        "*.test.tsx",
      ],
      "parserOptions": {
        "project": [
          "packages/react-native-app/tsconfig.spec.json",
        ],
      },
    },
    {
      "files": [
        "*.stories.tsx",
      ],
      "parserOptions": {
        "project": [
          "packages/react-native-app/tsconfig.storybook.json",
        ],
      },
    },
    {
      "excludedFiles": [
        "jest.config.ts",
        "test/**/*.ts",
        "*.spec.ts",
        "*.test.ts",
        "test/**/*.tsx",
        "*.stories.tsx",
        "*.spec.tsx",
        "*.test.tsx",
      ],
      "files": [
        "*.ts",
        "*.tsx",
      ],
      "parserOptions": {
        "project": [
          "packages/react-native-app/tsconfig.app.json",
        ],
      },
    },
    {
      "files": [
        "*.tsx",
      ],
      "rules": {
        "react/react-in-jsx-scope": "off",
      },
    },
  ],
}
`;

exports[`reactNativeAppGenerator Gemfile Gemfile matches snapshot 1`] = `
"source "https://rubygems.org"

# You may use http://rbenv.org/ or https://rvm.io/ to install and use this version
ruby ">= 3.2.2"

gem "cocoapods", "~> 1.12"
"
`;

exports[`reactNativeAppGenerator e2e .detoxrc.json matches snapshot 1`] = `
"{
  "testRunner": {
    "args": {
      "$0": "jest",
      "config": "./jest.config.json"
    },
    "jest": {
      "setupTimeout": 120000
    }
  },
  "apps": {
    "ios.debug": {
      "type": "ios.app",
      "build": "cd ../../packages/react-native-app/ios && xcodebuild -workspace ReactNativeApp.xcworkspace -scheme ReactNativeApp -configuration Debug -sdk iphonesimulator -destination 'platform=iOS Simulator,name=iPhone 14' -derivedDataPath ./build -quiet",
      "binaryPath": "../../packages/react-native-app/ios/build/Build/Products/Debug-iphonesimulator/ReactNativeApp.app"
    },
    "ios.release": {
      "type": "ios.app",
      "build": "cd ../../packages/react-native-app/ios && xcodebuild -workspace ReactNativeApp.xcworkspace -scheme ReactNativeApp -configuration Release -sdk iphonesimulator -destination 'platform=iOS Simulator,name=iPhone 14' -derivedDataPath ./build -quiet",
      "binaryPath": "../../packages/react-native-app/ios/build/Build/Products/Release-iphonesimulator/ReactNativeApp.app"
    },
    "android.debug": {
      "type": "android.apk",
      "build": "cd ../../packages/react-native-app/android && ./gradlew assembleDebug assembleAndroidTest -DtestBuildType=debug",
      "binaryPath": "../../packages/react-native-app/android/app/build/outputs/apk/debug/app-debug.apk"
    },
    "android.release": {
      "type": "android.apk",
      "build": "cd ../../packages/react-native-app/android && ./gradlew assembleRelease assembleAndroidTest -DtestBuildType=release",
      "binaryPath": "../../packages/react-native-app/android/app/build/outputs/apk/release/app-release.apk"
    }
  },
  "devices": {
    "simulator": {
      "type": "ios.simulator",
      "device": {
        "type": "iPhone 14"
      }
    },
    "emulator": {
      "type": "android.emulator",
      "device": {
        "avdName": "Pixel_4a_API_30"
      }
    }
  },
  "configurations": {
    "ios.sim.release": {
      "device": "simulator",
      "app": "ios.release"
    },
    "ios.sim.debug": {
      "device": "simulator",
      "app": "ios.debug"
    },
    "android.emu.release": {
      "device": "emulator",
      "app": "android.release"
    },
    "android.emu.debug": {
      "device": "emulator",
      "app": "android.debug"
    }
  }
}
"
`;

exports[`reactNativeAppGenerator e2e .eslintrc.json matches snapshot 1`] = `
{
  "extends": [
    "../../.eslintrc.json",
  ],
  "ignorePatterns": [
    "node_modules",
  ],
  "overrides": [
    {
      "files": [
        "jest.config.ts",
        "test/**/*.ts",
        "*.spec.ts",
        "*.test.ts",
      ],
      "parserOptions": {
        "project": [
          "e2e/react-native-app-e2e/tsconfig.spec.json",
        ],
      },
    },
  ],
}
`;

exports[`reactNativeAppGenerator e2e jest.config.json jest.config.json matches snapshot 1`] = `
"{
  "preset": "../../jest.preset",
  "rootDir": ".",
  "testMatch": [
    "<rootDir>/test/**/*.test.ts?(x)",
    "<rootDir>/test/**/*.spec.ts?(x)"
  ],
  "testTimeout": 120000,
  "maxWorkers": 1,
  "globalSetup": "detox/runners/jest/globalSetup",
  "globalTeardown": "detox/runners/jest/globalTeardown",
  "reporters": [
    [
      "jest-junit",
      {
        "addFileAttribute": "true",
        "classNameTemplate": "{suitename}",
        "outputDirectory": "reports/junit",
        "outputName": "react-native-app-e2e.xml"
      }
    ]
  ],
  "testEnvironment": "detox/runners/jest/testEnvironment",
  "verbose": true,
  "setupFilesAfterEnv": [
    "<rootDir>/test/setup/setup.ts"
  ],
  "coverageDirectory": "../../reports/coverage/e2e/react-native-app-e2e"
}
"
`;

exports[`reactNativeAppGenerator e2e project.json matches snapshot 1`] = `
{
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "implicitDependencies": [
    "react-native-app",
  ],
  "name": "react-native-app-e2e",
  "projectType": "application",
  "sourceRoot": "packages/react-native-app-e2e/src",
  "tags": [],
  "targets": {
    "build-android": {
      "configurations": {
        "production": {
          "detoxConfiguration": "android.emu.release",
        },
      },
      "executor": "@nx/detox:build",
      "options": {
        "detoxConfiguration": "android.emu.debug",
      },
    },
    "build-ios": {
      "configurations": {
        "production": {
          "detoxConfiguration": "ios.sim.release",
        },
      },
      "executor": "@nx/detox:build",
      "options": {
        "detoxConfiguration": "ios.sim.debug",
      },
    },
    "lint": {
      "executor": "@nx/linter:eslint",
      "options": {
        "lintFilePatterns": [
          "e2e/react-native-app-e2e/**/*.{ts,tsx,js,jsx}",
        ],
      },
      "outputs": [
        "{options.outputFile}",
      ],
    },
    "test-android": {
      "configurations": {
        "production": {
          "buildTarget": "react-native-app-e2e:build-android:production",
          "detoxConfiguration": "android.emu.release",
        },
      },
      "executor": "@nx/detox:test",
      "options": {
        "buildTarget": "react-native-app-e2e:build-android",
        "detoxConfiguration": "android.emu.debug",
      },
    },
    "test-ios": {
      "configurations": {
        "production": {
          "buildTarget": "react-native-app-e2e:build-ios:production",
          "detoxConfiguration": "ios.sim.release",
        },
      },
      "executor": "@nx/detox:test",
      "options": {
        "buildTarget": "react-native-app-e2e:build-ios",
        "detoxConfiguration": "ios.sim.debug",
      },
    },
  },
}
`;

exports[`reactNativeAppGenerator e2e project.json matches snapshot 2`] = `
"{
  "name": "react-native-app-e2e",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "packages/react-native-app-e2e/src",
  "projectType": "application",
  "targets": {
    "build-ios": {
      "executor": "@nx/detox:build",
      "options": {
        "detoxConfiguration": "ios.sim.debug"
      },
      "configurations": {
        "production": {
          "detoxConfiguration": "ios.sim.release"
        }
      }
    },
    "test-ios": {
      "executor": "@nx/detox:test",
      "options": {
        "detoxConfiguration": "ios.sim.debug",
        "buildTarget": "react-native-app-e2e:build-ios"
      },
      "configurations": {
        "production": {
          "detoxConfiguration": "ios.sim.release",
          "buildTarget": "react-native-app-e2e:build-ios:production"
        }
      }
    },
    "build-android": {
      "executor": "@nx/detox:build",
      "options": {
        "detoxConfiguration": "android.emu.debug"
      },
      "configurations": {
        "production": {
          "detoxConfiguration": "android.emu.release"
        }
      }
    },
    "test-android": {
      "executor": "@nx/detox:test",
      "options": {
        "detoxConfiguration": "android.emu.debug",
        "buildTarget": "react-native-app-e2e:build-android"
      },
      "configurations": {
        "production": {
          "detoxConfiguration": "android.emu.release",
          "buildTarget": "react-native-app-e2e:build-android:production"
        }
      }
    },
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": [
        "{options.outputFile}"
      ],
      "options": {
        "lintFilePatterns": [
          "e2e/react-native-app-e2e/**/*.{ts,tsx,js,jsx}"
        ]
      }
    }
  },
  "tags": [],
  "implicitDependencies": [
    "react-native-app"
  ]
}
"
`;

exports[`reactNativeAppGenerator e2e typescript tsconfig.json tsconfig.json matches snapshot 1`] = `
"{
  "extends": "../../tsconfig.base.json",
  "files": [],
  "include": [],
  "references": [
    {
      "path": "./tsconfig.spec.json"
    }
  ]
}
"
`;

exports[`reactNativeAppGenerator e2e typescript tsconfig.spec.json matches snapshot 1`] = `
"{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "sourceMap": false,
    "outDir": "../../dist/out-tsc",
    "allowJs": true,
    "types": ["node", "jest", "detox"]
  },
  "include": ["test/**/*.ts", "test/**/*.js"]
}
"
`;

exports[`reactNativeAppGenerator jest.config.ts jest.config.ts matches snapshot 1`] = `
"export default {
  displayName: "react-native-app",
    preset: "react-native",
  resolver: "@nx/jest/plugins/resolver",
  setupFilesAfterEnv: ["<rootDir>/test/setup/setup.ts"],
  moduleNameMapper: {
    "\\\\.svg$": "@nx/react-native/plugins/jest/svg-mock",
  },
            moduleFileExtensions: ["ts", "tsx", "js", "jsx", "html", "json"],
    coverageDirectory: "../../reports/coverage/packages/react-native-app",
  reporters: [
    "default",
    [
      "jest-junit",
      {
        addFileAttribute: "true",
        classNameTemplate: "{suitename}",
        outputDirectory: "reports/junit",
        outputName: "react-native-app.xml",
      },
    ],
  ],
};
"
`;

exports[`reactNativeAppGenerator native projects fastlane Appfile Appfile matches snapshot 1`] = `undefined`;

exports[`reactNativeAppGenerator native projects fastlane Fastfile matches snapshot 1`] = `undefined`;

exports[`reactNativeAppGenerator native projects ios Podfile Podfile matches snapshot 1`] = `
"# Resolve react_native_pods.rb with node to allow for hoisting
require Pod::Executable.execute_command("node", ["-p",
  'require.resolve(
    "react-native/scripts/react_native_pods.rb",
    {paths: [process.argv[1]]},
  )', __dir__]).strip

platform :ios, min_ios_version_supported
prepare_react_native_project!

# If you are using a \`react-native-flipper\` your iOS build will fail when \`NO_FLIPPER=1\` is set.
# because \`react-native-flipper\` depends on (FlipperKit,...) that will be excluded
#
# To fix this you can also exclude \`react-native-flipper\` using a \`react-native.config.js\`
# \`\`\`js
# module.exports = {
#   dependencies: {
#     ...(process.env.NO_FLIPPER ? { "react-native-flipper": { platforms: { ios: null } } } : {}),
# \`\`\`
flipper_config = ENV["NO_FLIPPER"] == "1" ? FlipperConfiguration.disabled : FlipperConfiguration.enabled

linkage = ENV["USE_FRAMEWORKS"]
if linkage != nil
  Pod::UI.puts "Configuring Pod with #{linkage}ally linked Frameworks".green
  use_frameworks! :linkage => linkage.to_sym
end

target "ReactNativeApp" do
  config = use_native_modules!

  # Flags change depending on the env values.
  flags = get_default_flags()

  use_react_native!(
    :path => config[:reactNativePath],
    # Hermes is now enabled by default. Disable by setting this flag to false.
    :hermes_enabled => flags[:hermes_enabled],
    :fabric_enabled => flags[:fabric_enabled],
    # Enables Flipper.
    #
    # Note that if you have use_frameworks! enabled, Flipper will not work and
    # you should disable the next line.
    :flipper_configuration => flipper_config,
    # An absolute path to your application root.
    :app_path => "#{Pod::Config.instance.installation_root}/.."
  )

  target "ReactNativeAppTests" do
    inherit! :complete
    # Pods for testing
  end

  post_install do |installer|
    # https://github.com/facebook/react-native/blob/main/packages/react-native/scripts/react_native_pods.rb#L197-L202
    react_native_post_install(
      installer,
      config[:reactNativePath],
      :mac_catalyst_enabled => false
    )
    __apply_Xcode_12_5_M1_post_install_workaround(installer)
  end
end
"
`;

exports[`reactNativeAppGenerator project.json matches snapshot 1`] = `
{
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "name": "react-native-app",
  "projectType": "application",
  "sourceRoot": "packages/react-native-app/src",
  "tags": [],
  "targets": {
    "build-android": {
      "dependsOn": [
        "ensure-symlink",
        "sync-deps",
      ],
      "executor": "@nx/react-native:build-android",
      "options": {},
      "outputs": [
        "{projectRoot}/android/app/build/outputs/bundle",
        "{projectRoot}/android/app/build/outputs/apk",
      ],
    },
    "build-ios": {
      "dependsOn": [
        "ensure-symlink",
        "sync-deps",
        "pod-install",
      ],
      "executor": "@nx/react-native:build-ios",
      "options": {},
      "outputs": [
        "{projectRoot}/ios/build/Build",
      ],
    },
    "bundle-android": {
      "dependsOn": [
        "ensure-symlink",
      ],
      "executor": "@nx/react-native:bundle",
      "options": {
        "bundleOutput": "dist/packages/react-native-app/android/main.jsbundle",
        "entryFile": "src/main.tsx",
        "platform": "android",
      },
      "outputs": [
        "{options.bundleOutput}",
      ],
    },
    "bundle-ios": {
      "dependsOn": [
        "ensure-symlink",
      ],
      "executor": "@nx/react-native:bundle",
      "options": {
        "bundleOutput": "dist/packages/react-native-app/ios/main.jsbundle",
        "entryFile": "src/main.tsx",
        "platform": "ios",
      },
      "outputs": [
        "{options.bundleOutput}",
      ],
    },
    "ensure-symlink": {
      "executor": "@nx/react-native:ensure-symlink",
      "options": {},
    },
    "lint": {
      "executor": "@nx/linter:eslint",
      "options": {
        "lintFilePatterns": [
          "packages/react-native-app",
        ],
      },
      "outputs": [
        "{options.outputFile}",
      ],
    },
    "pod-install": {
      "executor": "@nx/react-native:pod-install",
      "options": {},
    },
    "run-android": {
      "dependsOn": [
        "ensure-symlink",
        "sync-deps",
      ],
      "executor": "@nx/react-native:run-android",
      "options": {},
    },
    "run-ios": {
      "dependsOn": [
        "ensure-symlink",
        "sync-deps",
        "pod-install",
      ],
      "executor": "@nx/react-native:run-ios",
      "options": {},
    },
    "serve": {
      "executor": "nx:run-commands",
      "options": {
        "command": "nx start react-native-app",
      },
    },
    "start": {
      "dependsOn": [
        "ensure-symlink",
        "sync-deps",
        "pod-install",
      ],
      "executor": "@nx/react-native:start",
      "options": {
        "port": 8081,
      },
    },
    "sync-deps": {
      "executor": "@nx/react-native:sync-deps",
      "options": {},
    },
    "test": {
      "configurations": {
        "ci": {
          "ci": true,
          "codeCoverage": true,
        },
      },
      "executor": "@nx/jest:jest",
      "options": {
        "jestConfig": "packages/react-native-app/jest.config.ts",
        "passWithNoTests": true,
      },
      "outputs": [
        "reports/coverage/packages/react-native-app",
        "reports/junit/react-native-app.xml",
      ],
    },
  },
}
`;

exports[`reactNativeAppGenerator tsconfig.app.json or tsconfig.lib.json matches snapshot 1`] = `
{
  "compilerOptions": {
    "declaration": true,
  },
  "exclude": [
    "jest.config.ts",
    "**/*.spec.ts",
    "**/*.test.ts",
    "**/*.spec.tsx",
    "**/*.test.tsx",
    "test",
  ],
  "extends": "./tsconfig.json",
  "include": [
    "**/*.ts",
    "**/*.tsx",
  ],
}
`;

exports[`reactNativeAppGenerator tsconfig.json matches snapshot 1`] = `
{
  "compilerOptions": {
    "jsx": "react-jsx",
    "lib": [
      "dom",
      "dom.iterable",
      "esnext",
    ],
    "module": "esnext",
    "outDir": "../../dist/out-tsc",
    "target": "es5",
  },
  "extends": "../../tsconfig.base.json",
  "files": [
    "../../node_modules/@nrwl/react/typings/cssmodule.d.ts",
  ],
  "include": [],
  "references": [
    {
      "path": "./tsconfig.app.json",
    },
    {
      "path": "./tsconfig.spec.json",
    },
  ],
}
`;

exports[`reactNativeAppGenerator tsconfig.spec.json matches snapshot 1`] = `
{
  "compilerOptions": {
    "module": "commonjs",
    "types": [
      "jest",
      "node",
    ],
  },
  "extends": "./tsconfig.json",
  "include": [
    "jest.config.ts",
    "**/*.spec.ts",
    "**/*.test.ts",
    "**/*.spec.tsx",
    "**/*.test.tsx",
    "**/*.d.ts",
    "test",
  ],
}
`;
